xapian-core/weight/tfidfweight.cc

```diff
@@ -49,10 +50,31 @@ TfIdfWeight::TfIdfWeight(const std::string &normals)
     need_stat(WDF_MAX);
 }
```

Comment:
You don't request the extra stats needed by one of the `P` normalisations if this constructor is used.